Frozen busybox latest glibc

arm64b opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines;
https://github.com/moby/moby/blob/master/CONTRIBUTING.md

** Make sure all your commits include a signature generated with `git commit -s` **

For additional information on our contributing process, read our contributing
guide https://docs.docker.com/opensource/code/

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->
As the follow-up of PR #36298, this PR:
1. Adding both `busybox:latest` and `busybox:glibc` into docker container as frozen images
2. Using `busybox:glibc` instead of `busybox:latest` as the testing image explicitly for those network related test cases.  

**- What I did**
Using `busybox:1.27-glibc` as the image to test network related test cases (`ping` issue)
**- How I did it**
Adding both `glibc` and `latest` tagged `busybox` into frozen images
**- How to verify it**
`make test-integration`
**- Description for the changelog**
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->


**- A picture of a cute animal (not mandatory but encouraged)**

vdemeester reviewed (approved):
LGTM :frog:

arm64b commented:
Thanks @vdemeester for the quick response 👍 !
Just checked the CI result, interesting, seems the service image `busybox` in the `simpleTestService()` have to be tagged by `latest`, otherwise the `CreateService()` cannot create the `busybox /bin/top` service. 
Because `TestAPISwarmHealthcheckNone` test case will `ping` sth, so we have to point the service image to `busybox: glibc` instead of `busybox:latest`, that will be a dilemma...
Will investigate further tomorrow. 
@vdemeester , @tianon @dnephin @thaJeztah  PTAL first?

arm64b commented on integration-cli/docker_api_swarm_test.go:
```diff
@@ -573,7 +573,7 @@ func simpleTestService(s *swarm.Service) {
 	s.Spec = swarm.ServiceSpec{
 		TaskTemplate: swarm.TaskSpec{
 			ContainerSpec: &swarm.ContainerSpec{
-				Image:   "busybox:latest",
```
mark

dnephin commented:
changes sgtm

tianon reviewed (approved):
LGTM

(any particular reason we're pinning explicitly to `1.27-glibc` instead of just `glibc`?)

arm64b commented:
The only issue we need to resolve for this PR is #36386

arm64b commented:
>(any particular reason we're pinning explicitly to 1.27-glibc instead of just glibc?)

No special reason, if any, maybe there're some testing noise in my local system prevent me from avoiding a little _big_ changes, so I'd like to get to the target gradually 🤗

thaJeztah reviewed (approved):
LGTM if green :smile:

arm64b commented:
@yongtang , would you pls help to rebuild the `experimental` node, if the same issue repeated, then we need to think about to fall back from `busybox:glibc` to `busbybox:1.27-glibc`, the reason why I use 1.27 in the first commit version is that the local testing result showed `busybox:glibc` is not stable (sometimes it will have issues, sometime it's not, that what I said it "testing noise".

arm64b commented:
Now rebuild should be OK, every maintainer please go ahead to rebuild it. Then we can decide the next step

arm64b committed:
Add `busybox:latest` into the frozen images

Adding `busybox:latest` and `busybox:glibc` as the frozen images

Signed-off-by: Dennis Chen <dennis.chen@arm.com>

arm64b committed:
Network testing with `busybox:glibc`

Using the `busybox:glibc` instead of `busybox:latest` to the
network related test cases (`ping` issue).

Signed-off-by: Dennis Chen <dennis.chen@arm.com>

arm64b commented:
Hello @vdemeester , I dig into the CI failure cases last night and make some little tweak for the codes today, now from the CI result, I believe the failure on node `powerpc` and `z` are testing noise:
`powerpc`: `...............................................no space left on device" 
`z`: only `DockerSwarmSuite.TearDownTest` fails

So I bet this time if you rebuild the 2 nodes, we will win 🤗, else I owe you a bottle of beer🐸

arm64b commented:
Oops, seems I lost a bottle of beer but actually not, because we human beings think the `powerpc` machine is not reliable recently 😈 This point can be backed by the CI result of the PRs submitted these days...

arm64b commented:
OK, thanks @thaJeztah , apparently you are a lucky man, now it's green 😺

thaJeztah commented:
Can't take credits for that; I think the test-infra team and @vdemeester have been cleaning up nodes that were running out of disk space 🤗

arm64b commented:
Ah, a very tough CI battle we have experienced together🤗, and from the CI I found we still have some tear-down steps in CI need to be enhanced, so it's not bad thing 👍
